accepted/0000-buildpack-yml.md

```diff
@@ -0,0 +1,130 @@
+# Buildpack YAML specification
+
+## Summary
+The `buildpack.yml` file should conform to a specified interface. The specification that follows provides both a generic schema and descibed the planned future format for buildpacks currently in the `paketo-buildpacks` and `paketo-community` orgs.
+
+## Motivation
+The `buildpack.yml` file is a public API that users will want to fit a rational and guessable form. Aligning all Paketo buildpacks with this form will help to rationalize this API. Additionally, it enables  users to guess where certain configuration settings might be defined, leading to a smoother feature discovery process.
+
+## Detailed Explanation
+We propose that the `buildpack.yml` file conform to the following schema.
+
+### Schema
+```yaml
+<language-family>:
+  <implementation>:
+    <key>: <value>
+```
+
+All configuration is grouped under a language-family namespace, and then logically grouped under an implementation buildpack namespace which "owns" that configuration. Naturally, buildpacks that do not "own" that configuration should also be able to rely upon this schema to discover settings for other parts of the build process.
+
+For example, the PHP language family consumes both `nginx` and `httpd` as web servers. The `php-web` buildpack would need a mechanism to discover which web server the user desires to use in their application so that it can configure PHP correctly for that web server. If these fields are standardized, then the `php-web` buildpack can detect that `nginx` is the server chosen if there is a `web-server.nginx` key in the `buildpack.yml` file.
+ 
+
+## Rationale and Alternatives
+
+The rationale for choosing this format is that language-family and implementation namespaces are currently the only mechanism for grouping configuration settings that can be consumed by buildpacks.
+
+An alternative to this type of grouping would be to allow buildpack authors to choose any mechanism that fits their own needs. Those systems, being unstandardized, would be difficult to integrate and likely lead to incompatibilities of buildpacks that may rely upon the same keys for different configuration settings.
+
+### Decision Points
+
+#### Should all keys be listed under a language family?
+It logically groups configuration for a buildpack family and reduces top level key pollution.
+
```

Comment:
Since language-family/meta-cnbs are abstract ordered lists, can't anybody write a meta cnb with their own name? In that case, it seems like an implementation buildpack would have no awareness of which "language-family" section its data sits under.
If the cnb searches the entire tree to find the relevant key, there's a chance of duplication and the burden of conflict resolution. How about a shallower tree with buildpack-ids (or whatever is unique) being the first level keys?